pkg/cmd/server/api/validation/validation.go

```diff
@@ -78,10 +78,10 @@ func ValidateCertInfo(certInfo api.CertInfo, required bool, fldPath *field.Path)
 
 	if required || len(certInfo.CertFile) > 0 || len(certInfo.KeyFile) > 0 {
 		if len(certInfo.CertFile) == 0 {
-			allErrs = append(allErrs, field.Required(fldPath.Child("certFile"), ""))
+			allErrs = append(allErrs, field.Required(fldPath.Child("certFile"), "Both certificate and key must be provided"))
```

Comment:
Sorry, I don't understand what you mean by "build the field path for the other element and specify it here." Are you suggesting I throw an error on both fields if either is specified without the other?

Yeah, I'll at least rephrase it to be "both or neither".